src/main/java/bio/terra/cda/app/controller/QueryApiController.java

```diff
@@ -95,12 +96,34 @@ private String createNextUrl(String jobId, int offset, int limit) {
   }
 
   @Override
-  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body) {
-
+  public ResponseEntity<QueryCreatedData> uniqueValues(String version, String body, String system) {
     String table = applicationConfiguration.getBqTable() + "." + version;
     NestedColumn nt = NestedColumn.generate(body);
-    String querySql = "SELECT DISTINCT " + nt.getColumn() + " FROM " + table + nt.getUnnestClause();
-    logger.debug("uniqueValues: " + querySql);
+    Set<String> unnestClauses = nt.getUnnestClauses();
+    final String whereClause;
+
+    if (system != null) {
+      NestedColumn whereColumns = NestedColumn.generate("ResearchSubject.identifier.system");
+      whereClause = " WHERE " + whereColumns.getColumn() + " = '" + system + "'";
+      // add any additional 'where' unnest partials that aren't already included in columns-unnest
+      // clauses
+      unnestClauses.addAll(whereColumns.getUnnestClauses());
+    } else {
+      whereClause = "";
+    }
+    StringBuffer unnestConcat = new StringBuffer();
+    unnestClauses.stream().forEach((k) -> unnestConcat.append(k));
+
+    String querySql =
+        "SELECT DISTINCT "
+            + nt.getColumn()
+            + " FROM "
+            + table
+            + unnestConcat
+            + whereClause
+            + " ORDER BY "
```

Comment:
Maybe I asked this before, but I don't know why the query uses `ORDER BY`. It seems like a caller could sort the results if they wanted ordered data.